packages/node_modules/@cerebral/baobab/src/index.js

```diff
@@ -0,0 +1,85 @@
+import Baobab from 'baobab'
+import BaseModel from 'cerebral/lib/BaseModel'
+
+export default class BaobabModelFactory {
+  constructor(baobabOptions = {}) {
+    class BaobabModel extends BaseModel {
+      constructor(controller) {
+        super(controller)
+        this.state = new Baobab(this.initialState)
+      }
+      get(path = []) {
+        return this.state.get(path)
+      }
+      set(path, value) {
+        this.state.set(path, value)
+        this.changedPaths.push({
+          forceChildPathUpdates: true,
+          path,
+        })
+      }
+      toggle(path) {
+        this.state.set(path, !this.state.get(path))
+        this.changedPaths.push({
+          path,
+        })
+      }
+      push(path, value) {
+        this.state.push(path, value)
+        this.changedPaths.push({
+          path,
+        })
+      }
+      merge(path, ...values) {
+        values.forEach(value => {
+          this.state.merge(path, value)
+          for (let prop in value) {
+            this.changedPaths.push({
+              forceChildPathUpdates: true,
+              path: path.concat(prop),
+            })
+          }
+        })
+      }
```

Comment:
With merge, had to add a check if value exists to mimmic behaviour of standard model

```js
merge(path, ...values) {
    if (!this.state.exists(path)) {
      this.state.set(path, {})
    }
    values.forEach(value => {
      this.state.merge(path, value)
      for (let prop in value) {
        this.changedPaths.push({
          forceChildPathUpdates: true,
          path: path.concat(prop),
        })
      }
    })
  }
```

Reply:
Okay, updated :)